httpserver/httpserver.go

```diff
@@ -0,0 +1,85 @@
+// Package httpserver handle the logic of configuring an HTTP server with TLS
+// certs, choosing a port to listen on, etc, which is orthogonal to the business
+// logic of the actual service.
+package httpserver
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+)
+
+// Config captures the http server related configuration from the environment.
+type Config struct {
+	ListenAddr string `env:"LISTEN_ADDR,required"`
+	Insecure   bool   `env:"INSECURE" envDefault:"false"`
+	TLSCert    string `env:"TLS_CERT"`
+	TLSKey     string `env:"TLS_KEY"`
+}
+
+// Validate the config, returning an error describing any problems which occur.
+func (config *Config) Validate() error {
+	host, _, err := net.SplitHostPort(config.ListenAddr)
+	if err != nil {
+		return fmt.Errorf("Error parsing ListenAddr: %v", err)
+	}
+
+	hostIP := net.ParseIP(host)
+	// if host was a hostname rather than an ip address, hostIP will be nil,
+	// so this will correctly evaulate to false:
+	isLoopback := hostIP.Equal(net.ParseIP("127.0.0.1")) || hostIP.Equal(net.ParseIP("::1"))
+
+	haveCert := config.TLSCert != ""
+	haveKey := config.TLSKey != ""
+
+	if config.Insecure && haveCert {
```

Comment:
Since we're doing config from environment variables now, should we remind people to delete (or make "") variables that would cause `Validate()` to error out? I feel like deleting lines from a config file is obvious, but it might not be too obvious that you have to remove env variables.

Reply:
Good call. I updated the error messages. I also found a couple places where we were using the old option names (e.g. `ListenAddr` instead of `LISTEN_ADDR`) and fixed them.